locksmith/src/websub/helpers/renewKey.ts

```diff
@@ -0,0 +1,138 @@
+import { ethers, constants, BigNumber } from 'ethers'
+import { Web3Service } from '@unlock-protocol/unlock-js'
+import networks from '@unlock-protocol/networks'
+
+import { KeyRenewal } from '../../models'
+import KeyPricer from '../../utils/keyPricer'
+import Dispatcher from '../../fulfillment/dispatcher'
+
+interface RenewKeyParams {
+  keyId: number
+  lockAddress: string
+  network: number
+}
+
+interface ShouldRenew {
+  shouldRenew: boolean
+  gasRefund?: string
+  error?: string
+}
+
+// precision base for gas calculations
+const BASE = 1000000
+
+const getGasFee = async (network: number) => {
+  const pricer = new KeyPricer()
+  const gasPrice = await pricer.gasFee(network, BASE)
```

Comment:
This function is currently computing the gas fee using the hardcoded gas cost/limit for a purchase. It would be useful to change how this function works so it can take the gas cost/limit as a param so we can re-use it.

Ideally we should also remove the "hardcoded" gas cost...